cmd/k8s-operator/ingress.go

```diff
@@ -290,11 +290,110 @@ func validateIngressClass(ctx context.Context, cl client.Client) error {
 	return nil
 }
 
+// checkPathConflict checks if a new path conflicts with existing paths.
+// For PathTypePrefix, a conflict occurs if one path is a prefix of another.
+// For PathTypeExact, a conflict occurs if the paths are identical.
+// For mixed types, a conflict occurs only if the prefix path is a prefix of the exact path.
+// Returns true if there's a conflict, along with the conflicting path.
+func checkPathConflict(newPath string, newPathType networkingv1.PathType, existingPaths map[string]networkingv1.PathType) (bool, string) {
+	// If the new path is empty, default it to "/"
+	if newPath == "" {
+		newPath = "/"
+	}
+
+	if !strings.HasPrefix(newPath, "/") {
+		newPath = "/" + newPath
+	}
+
+	for existingPath, existingPathType := range existingPaths {
+		// For exact path types, only identical paths conflict
+		if newPathType == networkingv1.PathTypeExact && existingPathType == networkingv1.PathTypeExact {
+			if newPath == existingPath {
+				return true, existingPath
+			}
+			continue
+		}
+
+		// For prefix path types or mixed types, check for prefix conflicts
+		if newPathType == networkingv1.PathTypePrefix || existingPathType == networkingv1.PathTypePrefix {
+			if newPathType != existingPathType {
+				if newPathType == networkingv1.PathTypeExact && existingPathType == networkingv1.PathTypePrefix {
```

Comment:
At the moment the operator treats every ingress resource it handles as a `PathTypePrefix`, or `PathTypeImplementationSpecific`. Given this, `PathTypeExact` isn't supported (see [here](https://github.com/tailscale/tailscale/blob/main/cmd/k8s-operator/ingress.go#L347-L348).